pars-derive/src/lib.rs

```diff
@@ -174,7 +178,50 @@ fn generate_fmt_block(
     attrs: &AttributeArgs,
     cont: &Container,
 ) -> Result<TokenStream, Vec<syn::Error>> {
-    unimplemented!()
+    let fn_ident = syn::Ident::new(
+        &format!("_PARS_matches_for_{}", cont.ident.to_string()),
+        Span::call_site(),
+    );
+    if let Data::Enum(_) = cont.data {
+        panic!("pars::fmt only works with structs");
+    }
+
+    let field_names = cont.data.all_fields().filter_map(|f| match &f.member {
+        syn::Member::Named(ident) => Some(ident.to_string()),
+        _ => None,
+    }).collect::<Vec<_>>();
+    let fmt_string = get_attr_string(attrs).map_err(|e| vec![e])?;
+    let num_fields = cont.data.num_fields();
+
+    let _ = ::pars_fmt::FmtMatcher::new(&fmt_string, field_names.as_slice()).unwrap();
```

Comment:
What's this about?

Reply:
just validation of the fmt string at compile time, we do the same thing with the regex.

Reply:
ah, got it. Thanks for the comment.